contents/blog/decouple-deployment-from-release.md

```diff
@@ -0,0 +1,114 @@
+
+2. It breaks other features or the entire app. For example, values created by new features corrupt other data unexpectedly and crashes the app in production.
+
+By preventing new code from automatically releasing after a deployment, you lower the risk of these issues. This is known as **decoupling deployment from release**.
+
+Doing this enables [testing code in production](/blog/testing-in-production). This process helps you discover issues arising from production infrastructure and the new code's integration with the real release.
+
+On top of lowering the risk of issues, decoupling can increase developer productivity and app reliability. The specific benefits include:
+
+- Improves developer confidence knowing they can test in production and that there is an easy way to roll back changes without affecting users.
+
+- Increase deployment frequency by removing the need to wait for a formal release. 
+
+- Developers can follow [trunk-based development](https://trunkbaseddevelopment.com/) and merge to `main` more often instead of waiting on branches. This increases shipping velocity and reduces merge conflicts.
+
+- Developers can pass off releases to product managers and marketers once they deploy. They can coordinate elaborate launches and stakeholders without holding up development.
+
+- Less downtime from fewer broken releases. Easier rollbacks when an issue does occur because infrastructure (like [feature flags](/feature-flags)) is already in place.
+
+## How we decouple deployment from release at PostHog
+
+As an example of decoupling deployment from release, we can look at what we do at PostHog. We deploy many changes behind feature flags and release them after testing them in production. At any time, there are [30+ flags](https://github.com/PostHog/posthog/blob/03eb1dcaec3cf5064a1ace4433f2f77d6676b634/frontend/src/lib/constants.tsx#L118C1-L118C1) in use in PostHog. They are created and updated daily.
+
+![PostHog's flags](../images/blog/decouple-deployment-from-release/flags.png)
+
+These flags help us dogfood features, roll out changes to specific users, do incremental rollouts, and more. For example, in the flags above, we deployed without releasing:
+
+- Loading person data from ClickHouse to improve performance.
+- Product-specific onboarding.
+- A frontend for the new batch export system.
+- Additional person and group options on dashboards for our customer success team.
+
+We also dark launch features such as our upgraded batch exports system and data warehouse. This might mean they are only accessible through the API, don’t have navigation links, or only have limited options in the released code.
+
+Each of these changes runs through an extensive testing suite locally and on pull requests. This includes unit, integration, end-to-end, and visual tests. This enables us to be confident when deploying in the first place.
```

Comment:
Not sure we need this. Feels a bit out of place: We're talking about how to safely release code in the above paragraphs, but then this section is talking about deploying safely all of a sudden
```suggestion
```